src/duplicacy_b2client.go

```diff
@@ -299,6 +323,18 @@ func (client *B2Client) ListFileNames(startFileName string, singleFile bool, inc
 
 		output := B2ListFileNamesOutput{}
 
+		if !inputMapBool && singleFile && !includeVersions {
```

Comment:
Again here you need to check if responseHeader is nil.

Reply:
I'll add that and some other error checking to be sure the expected headers are present.